### ISSUE TITLE
Login with SAML (Windows ADFS) in the Application for Mac OS doesn't work

### ISSUE THREAD

**Issue by webr0ck** [mattermost/desktop#2509]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: https://docs.mattermost.com/install/troubleshooting.html](https://docs.mattermost.com/install/troubleshooting.html).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**

I can't login with SAML in the MacOs application. All works for browsers, windows or mobile apps.

In the app logs, I see a request to the ADFS server but nothing more. No requests from App to the MM server

From the app logs:

```
[2023-01-20 14:53:44.248] [debug] webContentEvents.did-start-navigation {
  webContentsId: 5,
  url: 'https://mm._server_/login/sso/saml?extra=expired'
}
[2023-01-20 14:53:44.249] [debug] webContentEvents.will-navigate {
  webContentsId: 5,
  url: 'https://mm._server_/login/sso/saml?extra=expired'
}
[2023-01-20 14:53:44.721] [debug] Utils.flushCookiesStore
[2023-01-20 14:53:46.055] [verbose] AuthManager.handleAppLogin {
  request: {
    url: 'https://_our_adfs_server_/adfs'
  }
```

**Environment**
- Operating System: MacOS
- Mattermost Desktop App version: 5.2.2 
- Mattermost Server version: 7.1.2.7.1.2

**Comment by webr0ck** [comment]
I click "SSO login" button and nothing happens (MacBook pro 16, 2019 with Intel CPU)

Additionally, we checked it on MacBook with ARM and an old MacBook with intel but with Catalina OS. SSO work for old Mac and for Mac with ARM cpu

**Comment by devinbinnie** [comment]
> I click "SSO login" button and nothing happens (MacBook pro 16, 2019 with Intel CPU)
> 
> Additionally, we checked it on MacBook with ARM and an old MacBook with intel but with Catalina OS. SSO work for old Mac and for Mac with ARM cpu

@webr0ck When you tested with the other machines, were they also using v5.2.2?

**Comment by webr0ck** [comment]
@devinbinnie yes, all the time this version

**Comment by webr0ck** [comment]
Update. SAML works in the case of:
MM Client from git (not from App Store)
MacOS version: Ventura 13.2.1

**Comment by webr0ck** [comment]
Update. This problem is caused WIA
MatterMost clients used ```User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.5249.119 Electron/21.2.0 Safari/537.36 Mattermost/5.2.2```
And you can fix it on the ADFS side, by creating a filter. For example, the rule can be next:
```
WIASupportedUserAgents
----------------------
{=~Chrome\S*\sSafari\S*$, =~Windows\sNT.*Edg.*, Edg/}
```

But we still wait for a new version, with native browser authentication.

**Comment by devinbinnie** [comment]
Relevant JIRA ticket: https://mattermost.atlassian.net/browse/MM-37984

**Comment by devinbinnie** [comment]
Hey all,

I want to apologize for the lack of communication on these tickets regarding authentication. But I come with some good news: we've been working on separating the external login flow from the Desktop App, deferring to the browser like many other applications do. This should overall improve security and stability around the login process, as the current implementation requires a few hacks to make it work correctly. 

As of today, we released Mattermost v9.1 which has this feature implemented, and should work with all existing Desktop App versions going back to at least v5.4.0. Going forward we will be supporting this login flow for all external providers using the Desktop App, and this should fix any issues around login flow. Your server will need to be upgraded to take advantage of this feature.

I'll be closing these tickets for now as fixed, but feel free to comment and ask questions if you have any. Thanks for your patience :)
